src/compiler/checker.ts

```diff
@@ -8402,22 +8402,28 @@ namespace ts {
                         return globalFunctionType;
                     case "Array":
                     case "array":
-                        return !typeArgs || !typeArgs.length ? anyArrayType : undefined;
+                        return !noImplicitAny && (!typeArgs || !typeArgs.length) ? anyArrayType : undefined;

```

Comment:
I'm not sure this is the right thing for JS. Maybe an implicit any error instead? Can you check on new errors in the user test suite? None (or few) have noImplicitAny on, so to make that easier you'll probably want to manually force noImplicitAny on in externalTestRunner.ts.